tests/test_trtllm_gen_decode.py

```diff
@@ -137,9 +137,19 @@ def test_trtllm_batch_decode_fmha(
     }
 
     sm_scale = float(1.0 / (head_dim**0.5))
-    q = torch.randn(batch_size, num_qo_heads, head_dim, device=device).to(
-        dtype_map[q_dtype]
-    )
+    if q_dtype == "fp8":
+        q = torch.randn(
+            batch_size, num_qo_heads, head_dim, dtype=torch.bfloat16, device=device
+        )
+        q, q_scale = to_float8(q)
+        # Reference implementation have functional issue or low precision with fp8, use bfloat16 and fake-quantization instead.
```

Comment:
This is mainly because reference implementation uses fp8 as output data type and it results in low precision, I'll fix the behavior and yes we can use bfloat16 as reference here instead.